src/jit/lsraxarch.cpp

```diff
@@ -2314,6 +2314,18 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
         }
     }
 
+    if ((op2 == nullptr) && (info->srcCount > 2) && node->isRMWHWIntrinsic(compiler))
```

Comment:
There are some nodes, such as `NI_SSE41_Insert` or `NI_SSE41_MultipleSumAbsoluteDifferences`, which have RMW semantics and more than two operands.

This currently handles that case with a TODO in order to track merging the logic back into `LinearScan::BuildNode`. When that happens, we should also update it to set `isTgtPref` for nodes with more than two arguments.

Reply:
NOTE: This is basically the logic removed in https://github.com/dotnet/coreclr/pull/16808/files#diff-498fd4859cb147c9bf082f5ebf32ca8fL2312, but changed to only apply to nodes with more than 2 arguments (since 1 and 2 argument nodes are being handled by the "common" code in `BuildNode`)